text/0000-input-macros.md

```diff
@@ -0,0 +1,484 @@
+        match result {
+            Ok(mut prices) => sum += prices,
+            Err(InputError::Eof) => {
+                println!("Oh no, EOF!");
+                break;
+            }
+            Err(e) => {
+                println!("Error: {:?}", e); // e could be a ParseFloatError
+                                            // or an IO error
+                return Err(e);
+            }
+        }
+    }
+    println!("{}", sum);
+    Ok(())
+}
+```
+
+## What is the impact of not doing this?
+If this RFC is not accepted, the current way of reading input from the user
+will remain the same. This means that new Rustaceans will have to learn how to
+use the `std::io::Stdin::read_line` function and how to handle the `Buffer`
+and the `String` types. This can be a barrier to entry for new Rustaceans and
+it can make the language less friendly. 
+
+This RFC was presented as a pre-RFC at a [Rust Argentina meetup](https://rust-lang.ar/p/2025-april/), 
+where it received positive feedback, particularly from attendees new to Rust 
+(many with backgrounds in NodeJS, Python, and Go). They found the current 
+approach to reading input in Rust complex and not very user-friendly, and were 
+enthusiastic about the proposed macros.
+
+They were not too happy with the current way of reading input from the user. 
+They think that it was too complex and not too friendly. 
+
+The presentation can be found [here](https://youtu.be/CjZq93pzOkA?t=4080) (in 
+Spanish)
+And yes, the guy with the black shirt with the Rust logo is me (👋).
+
+## Could this be done in a library or macro instead? 
+
+Well yes, but I think that this is a good idea to have it in the standard
+library. I think that this is a good idea to have it in the standard library
+because it is a common use case and it is a good idea to have it in the
+standard library to make the language more friendly.
+
+Another way that we could consider this is added like a pseudo oficial library 
+like `rand` which is not in the standard library but is recommended by the Rust 
+team and the oficial documentation as the book does it.
+
+# Prior art
+[prior-art]: #prior-art
+
+Python has `input()`, Ruby has `gets`, C# has `Console.ReadLine()`
+... all of these return a string read from standard input.
+
+Some behaviors are different, for example:
+- Python's `input()` returns a string, but if the input is EOF
+  it raises an `EOFError` exception.
+- Ruby's `gets` returns `nil` if the input is EOF.
+- C#'s `Console.ReadLine()` returns `null` if the input is EOF.
+- Go's `bufio.Reader.ReadString()` returns an error if the input is EOF.
+
+Maybe a thing to have in mind is that in JavaScript we have the `prompt` function
+but this function in the case of the browser enable a little dialog when we 
+select cancel this dialog we receive a `null` value.
+This is a [specification in JavaScript](https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#dom-prompt-dev).
```

Comment:
Remark (a stretch tbh): note that Python 2 (and 1 too I believe) `input()` did `eval(raw_input(prompt))`, with the user input parsed and evaluated as a Python 2 expression might be arguably "closer" to the proposed `input!()` (with the proposed `input!()` being less of a footgun, of course, since the user would have more control over what error can occur). Python 3 changed it to just return a string.

Reply:
plus no arbitrary code execution...if you can type `__import__("os").execlp("bash", "bash")` as a reply to a `input` prompt and get it to execute a shell, your program is about as insecure as you can get

Reply:
One of the design goals is precisely to offer an ergonomic experience without falling into security issues or footguns like the old `eval`.